packages/app/src/components/swap/swap.tsx

```diff
@@ -92,10 +95,17 @@ const Swap: FC<Props> = ({
         }
     ) ?? [];
 
-  const handleSendTokenInput = (e: any) => {
+  const handleSendTokenInput: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
     setSendTokenAmount(e.target.value);
-    setReceiveTokenAmount(calcOutGivenIn);
-  };
+  }, []);
+
+  useEffect(() => {
```

Comment:
In this case you might just as well use the `calcOutGivenIn` value from the hook directly in the receive field without a `useEffect` as is in L167, no ?